src/storage/compression/bitpacking.cpp

```diff
@@ -20,14 +21,47 @@ static constexpr const idx_t BITPACKING_WIDTH_GROUP_SIZE = 1024;
 
 struct EmptyBitpackingWriter {
 	template <class T>
-	static void Operation(T *values, bool *validity, bitpacking_width_t width, idx_t count, void *data_ptr) {
+	static void Operation(T *values, bool *validity, bitpacking_width_t width, T frame_of_reference, idx_t count,
+	                      void *data_ptr) {
+	}
+};
+
+template <class T>
+struct BufferMinMax {
```

Comment:
I think this does not need to be a separate Struct. 

I think this would be simplified by removeing the `BufferMinMax` struct by having the bitpacking state contain a `T min`, `T max` and `bool min_max_set` property.  Then a `TryUpdateMinMax(T value)` combined with a `ResetMinMax()` that is called from the Flush.

Also it seems a bit cleaner to first do TrySubtractOperator, and then updating the min/max values when this succeeds because now a `TryUpdate` that fails, still sets the failing min/max values.